src/modules/ekf2/EKF/airspeed_fusion.cpp

```diff
@@ -150,31 +109,24 @@ void Ekf::fuseAirspeed(estimator_aid_source_1d_s &airspeed)
 		return;
 	}
 
-	const float HK9 = 1.0F/(innov_var);
-
 	_fault_status.flags.bad_airspeed = false;
 
-	// Observation Jacobians
-	SparseVector24f<4,5,6,22,23> Hfusion;
-	Hfusion.at<4>() = HK4;
-	Hfusion.at<5>() = HK5;
-	Hfusion.at<6>() = HK6;
-	Hfusion.at<22>() = HK7;
-	Hfusion.at<23>() = HK8;
+	Vector24f H; // Observation jacobian
+	Vector24f K; // Kalman gain vector
+
+	matrix::Vector<float, 24> state_vector(reinterpret_cast<const float*>(&_state));
```

Comment:
This is still a copy into state_vector.

I would just do the dumb/simple/safe/verbose version for now until we're ready to make a more sweeping change. 

state_vector(0) = _state.quat_nominal(0), etc

Reply:
It's actually already implemented: https://github.com/PX4/PX4-Autopilot/blob/20b46ad33f0a64ad6beabe5a8ab2a27fd3bdee9b/src/modules/ekf2/EKF/ekf_helper.cpp#L624-L636